addons/mrp/models/mrp_production.py

```diff
@@ -1748,7 +1748,16 @@ def _check_sn_uniqueness(self):
                     duplicates_unbuild = self.env['stock.move.line'].search_count(domain_unbuild + [
                         ('move_id.unbuild_id', '!=', False)
                     ])
-                    if not (duplicates_unbuild and duplicates - duplicates_unbuild == 0):
+                    # Maybe some components have been removed via a repair
+                    removed = 0
+                    if 'repair.line' in self.env:
```

Comment:
Mhhh the code of mrp shouldn't contains code of repair. 
In master, we have the module of mrp_repair but not in 14.0. It isn't possible to search on stock.move.line if there a SN has return in stock (instead of searching on repair) ?

Reply:
Alright I tried something else, let's see if the tests are okay with it

Reply:
I'll write a test to make sure though. I'm planning on writing it in repair, but having `if 'mrp.production' not in self.env: return` since repair does not depends on MRP, or if you know a better way to test only if the two (MRP and repair) modules are installed, please let me know :smile:

Reply:
@ryv-odoo the test is written :smile: I hope you'll like it